src/wp-includes/class-wp-theme-json.php

```diff
@@ -493,25 +545,83 @@ public function get_settings() {
 	 * the theme.json structure this object represents.
 	 *
 	 * @since 5.8.0
+	 * @since 5.9.0 Changed the arguments passed to the function.
 	 *
-	 * @param string $type Optional. Type of stylesheet we want. Accepts 'all',
-	 *                     'block_styles', and 'css_variables'. Default 'all'.
+	 * @param array $types    Types of styles to load. Will load all by default. It accepts:
+	 *                         'variables': only the CSS Custom Properties for presets & custom ones.
+	 *                         'styles': only the styles section in theme.json.
+	 *                         'presets': only the classes for the presets.
+	 * @param array $origins A list of origins to include. By default it includes self::VALID_ORIGINS.
 	 * @return string Stylesheet.
 	 */
-	public function get_stylesheet( $type = 'all' ) {
+	public function get_stylesheet( $types = array( 'variables', 'styles', 'presets' ), $origins = self::VALID_ORIGINS ) {
```

Comment:
What if someone is using this class in their own code and passing a string.

Reply:
This is a breaking change.

Reply:
Good point I guess we can easily map the old string parameter to the new array format and avoid the breaking change.